tap_agent/src/config.rs

```diff
@@ -0,0 +1,320 @@
+}
+
+#[derive(Clone, Debug, Args, Serialize, Deserialize, Default)]
+#[group(required = true, multiple = true)]
+pub struct NetworkSubgraph {
+    #[clap(
+        long,
+        value_name = "network-subgraph-deployment",
+        env = "NETWORK_SUBGRAPH_DEPLOYMENT",
+        help = "Network subgraph deployment"
+    )]
+    pub network_subgraph_deployment: Option<DeploymentId>,
+    #[clap(
+        long,
+        value_name = "network-subgraph-endpoint",
+        env = "NETWORK_SUBGRAPH_ENDPOINT",
+        default_value_t = String::from("https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-goerli"),
+        help = "Endpoint to query the network subgraph from",
+    )]
+    pub network_subgraph_endpoint: String,
+    #[clap(
+        long,
+        value_name = "allocation-syncing-interval",
+        env = "ALLOCATION_SYNCING_INTERVAL",
+        default_value_t = 120_000,
+        help = "Interval (in ms) for syncing indexer allocations from the network"
+    )]
+    pub allocation_syncing_interval: u64,
+}
+
+#[derive(Clone, Debug, Args, Serialize, Deserialize, Default)]
+#[group(required = true, multiple = true)]
+pub struct EscrowSubgraph {
+    #[clap(
+        long,
+        value_name = "escrow-subgraph-deployment",
+        env = "ESCROW_SUBGRAPH_DEPLOYMENT",
+        help = "Escrow subgraph deployment"
+    )]
+    pub escrow_subgraph_deployment: Option<DeploymentId>,
+    #[clap(
+        long,
+        value_name = "escrow-subgraph-endpoint",
+        env = "ESCROW_SUBGRAPH_ENDPOINT",
+        // TODO:
+        // default_value_t = String::from("https://api.thegraph.com/subgraphs/name/?????????????"),
+        help = "Endpoint to query the network subgraph from"
+    )]
+    pub escrow_subgraph_endpoint: String,
+    #[clap(
+        long,
+        value_name = "escrow-syncing-interval",
+        env = "ESCROW_SYNCING_INTERVAL",
+        default_value_t = 120_000,
+        help = "Interval (in ms) for syncing indexer escrow accounts from the escrow subgraph"
+    )]
+    pub escrow_syncing_interval: u64,
+}
+
+#[derive(Clone, Debug, Args, Serialize, Deserialize, Default)]
+#[group(required = true, multiple = true)]
+pub struct Tap {
+    #[clap(
+        long,
+        value_name = "rav-request-trigger-value",
+        env = "RAV_REQUEST_TRIGGER_VALUE",
+        help = "Value of unaggregated fees that triggers a RAV request (in GRT wei).",
+        default_value_t = 10_000_000_000_000_000_000 // 10 GRT
+    )]
+    pub rav_request_trigger_value: u64,
+    #[clap(
+        long,
+        value_name = "rav-request-timestamp-buffer",
+        env = "RAV_REQUEST_TIMESTAMP_BUFFER",
+        help = "Buffer (in ns) to add between the current time and the timestamp of the \
+        last unaggregated fee when triggering a RAV request.",
+        default_value_t = 1_000_000_000 // 1 second
+    )]
+    pub rav_request_timestamp_buffer_ns: u64,
+
+    // TODO: Remove this whenever the the gateway registry is ready
```

Comment:
how do we expect indexers to have this file? does senders register or host a discoverable aggregator endpoint somewhere? would you please provide an example file I could run the service with?

Reply:
It will be in the form of a smart contract. That's the current plan/suggestion I got from @Theodus . But no work started on this yet I believe.

Reply:
PS: the file is just to be able to get going for testing, basically.